src/main/java/teammates/storage/entity/Entity.java

```diff
@@ -0,0 +1,20 @@
+package teammates.storage.entity;
+
+import java.lang.reflect.Field;
+
+import javax.jdo.annotations.PrimaryKey;
+
+final class Entity {
+    private Entity() {
+    }
+
+    static String getFieldWithPrimaryKeyAnnotation(Class<?> cls) {
+        Field[] fs = cls.getDeclaredFields();
+        for (Field f : fs) {
+            if (f.isAnnotationPresent(PrimaryKey.class)) {
+                return f.getName();
+            }
+        }
+        return "";
```

Comment:
I know this was the example I gave, but can this actually be an assertion failure instead? e.g.
```java
Assumption.fail("There should be a field annotated with @PrimaryKey");
```